mailbox/api/src/test/java/org/apache/james/mailbox/manager/IntegrationResources.java

```diff
@@ -21,28 +21,28 @@
 
 import org.apache.james.mailbox.MailboxManager;
 import org.apache.james.mailbox.MessageIdManager;
-import org.apache.james.mailbox.quota.MaxQuotaManager;
-import org.apache.james.mailbox.quota.QuotaManager;
 import org.apache.james.mailbox.acl.GroupMembershipResolver;
 import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.quota.MaxQuotaManager;
+import org.apache.james.mailbox.quota.QuotaManager;
 import org.apache.james.mailbox.quota.QuotaRootResolver;
 
 /**
  * Provides empty resources for integration tests.
  */
-public interface IntegrationResources {
+public interface IntegrationResources<T extends MailboxManager> {
```

Comment:
do you really need parametric type here?

Reply:
Yes.

To instanciate the MessageIdManager I need a StoreMessageIdManager to get class internals (Quota, events, etc...). It is required as the class is stateless.

Reply:
I have some Eclipse warnings due to this, but I will not loose too much time on this for the moment.